src/Microsoft.Health.Fhir.Shared.Api/Modules/FhirModule.cs

```diff
@@ -80,8 +75,48 @@ ResourceElement SetMetadata(Resource resource, string versionId, DateTimeOffset
                     {
                         FhirResourceFormat.Json, (str, version, lastModified) =>
                         {
-                            var resource = jsonParser.Parse<Resource>(str);
-
+                            Resource resource = null;
+                            var parsed = false;
+                            do
+                            {
+                                try
+                                {
+                                    resource = jsonParser.Parse<Resource>(str);
+                                    parsed = true;
+                                }
+                                catch (StructuralTypeException ex)
+                                {
+                                    var match = MessageChecker.Match(ex.Message);
+                                    if (match.Success && match.Groups.Count == 4 && match.Groups["type"].Value == "date")
+                                    {
+                                        var valueToReplace = match.Groups["value"].Value;
+                                        var location = match.Groups["location"].Value;
+                                        var replace = valueToReplace.Substring(0, 10);
+                                        var root = FhirJsonNode.Parse(str, KnownResourceTypes.Resource);
+                                        var currentNode = root;
+                                        while (currentNode != null)
+                                        {
+                                            foreach (var child in currentNode.Children())
+                                            {
+                                                if (location.StartsWith(child.Location, StringComparison.OrdinalIgnoreCase))
+                                                {
+                                                    currentNode = child;
+                                                    break;
+                                                }
+                                            }
+
+                                            if (currentNode.Location == location)
+                                            {
+                                                break;
+                                            }
+                                        }
+
+                                        (currentNode as FhirJsonNode).JsonValue.Value = replace;
+                                        str = root.ToJson();
+                                    }
+                                }
+                            }
+                            while (!parsed);
```

Comment:
Final checkpoint, is there a chance we'll get into this loop and our modifications don't fix all the errors and it gets stuck?

Reply:
Put count up to 100, and add logging, but I doubt I can log json itself.